features/0067-didcomms-diddoc-conventions/README.md

```diff
@@ -0,0 +1,180 @@
+- priority : This represents the priority of the service endpoint, used for distinction when multiple `did-communication` service endpoints are present in a single Did document. It is mandatory that this field is set to an unsigned integer with the default value of `0`.
+- recipientKeys : This is an array of [did key references](https://w3c-ccg.github.io/did-spec/#public-keys) used to denote the default recipients of an endpoint. (*note-1*)
+- routingKeys: This is an array of [did key references](https://w3c-ccg.github.io/did-spec/#public-keys), ordered from most destward to most srcward, used to denote the individual routing hops in between the sender and recipients. See [TBC]() for more information on how routing is intended to operate.
+- serviceEndpoint : Required by the [Service Endpoints Spec](https://w3c-ccg.github.io/did-spec/#service-endpoints). This URL based endpoint is used to declare how the message should be sent. Did communication is transport agnostic, and therefore leverages existing application level transport protocols. However for each transport defined, which is identified by the URL scheme e.g `http`, a set of transport specific considerations are defined see [transports](../0025-didcomms-transports/README.md) for more details.
+
+>Notes
+>1. The keys featured in this array must resolve to keys of the same type, for example a mix `Ed25519VerificationKey2018` or `RsaVerificationKey2018` in the same array is invalid.
+
+### Message Preparation Conventions
+
+Below describes the process under which a Did communication message is prepared and sent to a Did based on the conventions declared in the associated Did document. The scenario in which the below is predicated has the following conditions.
+    - The sender possesses the Did document for the intended recipient(s) of a Did communication message.
+    - The sender has created a [content level message](../../concepts/0021-didcomm-message-anatomy/README.md) that is now ready to be prepared for sending to the intended recipient(s). 
+
+1. The sender resolves the relevant `did-communication` service of the intended recipient(s) Did document.
+2. The sender resolves the recipient keys present in the `recipientKeys` array of the service declaration.
+3. Using the resolved keys, the sender takes the content level message and packs it inside an [encrypted envelope](../../concepts/0021-didcomm-message-anatomy/README.md) for the recipient keys. (*note-2*)
+4. The sender then inspects the `routingKeys` array, if it is found to be empty, then the process skips to step 5. Otherwise, the sender prepares a content level message of type `forward`. The resolved keys from the `recipientKeys` array is set as the contents of the `to` field in the forward message and the encrypted envelope from the previous step is set as the contents of the `msg` field in the forward message. Following this, for each element in the `routingKeys` array the following sub-process is repeated:
```

Comment:
@tplooker I think the value of the `to` field should just be a DID with no fragment (ie. just the `"did:" method-name ":" method-specific-id` portion of the key reference)

Reply:
@llorllale the current thinking is that the to field is actually the raw public key value(s). Because when the routing identifier is a did or a did key reference, this implies the intermediary (or mediator agent is what we have previously referred to) needs to know the did. When the did is pairwise, this involves authenticated exchange of this did between the owner and the mediator. Practically this adds quite alot of overhead to how routing is conducted too. Rather when routing is based on public keys, it is far lighter and proving to a mediator you own the public key and hence should be able to create a route is far more light weight. I realize that without the routing RFC much of this is unclear so I will endeavor to get that into a proposal phase, language in this RFC can then also be cleaned up.

Reply:
@tplooker +1 for updating the routing RFC